tests/backup/backup_longevity_test.go

```diff
@@ -0,0 +1,409 @@
+	pureTopologyEnabled = false
+
+	//Default is allow deploying apps both in storage and storageless nodes
+	hyperConvergedTypeEnabled = true
+
+	// Pure Topology Label array
+	labels []map[string]string
+)
+
+// TriggerFunction represents function signature of a testTrigger
+type TriggerFunction func(*[]*scheduler.Context, *chan *EventRecord)
+
+var _ = Describe("{BackupLongevity}", func() {
+
+	IsBackupLongevityRun = true
+
+	contexts := make([]*scheduler.Context, 0)
+	var triggerLock sync.Mutex
+	var emailTriggerLock sync.Mutex
+	var populateDone bool
+	triggerEventsChan := make(chan *EventRecord, 100)
+	triggerFunctions = map[string]func(*[]*scheduler.Context, *chan *EventRecord){
+		CreatePxBackup:           TriggerCreateBackup,
+		CreatePxBackupAndRestore: TriggerCreateBackupAndRestore,
+		CreateRandomRestore:      TriggerCreateRandomRestore,
+		DeployBackupApps:         TriggerDeployBackupApps,
+	}
+	//Creating a distinct trigger to make sure email triggers at regular intervals
+	emailTriggerFunction = map[string]func(){
+		EmailReporter: TriggerEmailReporter,
+	}
+
+	BeforeEach(func() {
+		if !populateDone {
+			StartPxBackupTorpedoTest("BackupLongevityTest",
+				"Longevity Run For Backup", nil, 0, ATrivedi, Q1FY24)
+			populateIntervals()
+			//  populateDisruptiveTriggers()
+			populateDone = true
+		}
+	})
+
+	It("has to schedule app and introduce test triggers", func() {
+		log.InfoD("schedule apps and start test triggers")
+		watchLog := fmt.Sprintf("Start watch on K8S configMap [%s/%s]",
+			configMapNS, testTriggersConfigMap)
+
+		Step(watchLog, func() {
+			log.InfoD(watchLog)
+			watchConfigMap()
+		})
+
+		TriggerDeployBackupApps(&contexts, &triggerEventsChan)
+		TriggerAddBackupCluster(&contexts, &triggerEventsChan)
+		TriggerAddBackupCredAndBucket(&contexts, &triggerEventsChan)
+
+		var wg sync.WaitGroup
+		Step("Register test triggers", func() {
+			for triggerType, triggerFunc := range triggerFunctions {
+				log.InfoD("Registering trigger: [%v]", triggerType)
+				go backupEventTrigger(&wg, &contexts, triggerType, triggerFunc, &triggerLock, &triggerEventsChan)
+				wg.Add(1)
+			}
+		})
+		log.InfoD("Finished registering test triggers")
+		if Inst().MinRunTimeMins != 0 {
+			log.InfoD("Longevity Tests  timeout set to %d  minutes", Inst().MinRunTimeMins)
+		}
+
+		Step("Register email trigger", func() {
+			for triggerType, triggerFunc := range emailTriggerFunction {
+				log.InfoD("Registering email trigger: [%v]", triggerType)
+				go emailEventTrigger(&wg, triggerType, triggerFunc, &emailTriggerLock)
+				wg.Add(1)
+			}
+		})
+		log.InfoD("Finished registering email trigger")
+
+		CollectEventRecords(&triggerEventsChan)
+		wg.Wait()
+		close(triggerEventsChan)
+		Step("teardown all apps", func() {
+			for _, ctx := range contexts {
+				TearDownContext(ctx, nil)
+			}
+		})
+	})
+	JustAfterEach(func() {
+		defer EndTorpedoTest()
+		AfterEachTest(contexts)
+	})
+})
+
+func backupEventTrigger(wg *sync.WaitGroup,
```

Comment:
can we use same testTriggerfunction in longevity_test passing backup trigger functions

Reply:
This is same as longevity test, we copied the function from same and renamed as per backup trigger